[EDU-6369] feat: add edge connector reference (#1750)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/secure/edge-connector/edge-connector.mdx

#### Path

**Path** allows you to specify a custom path from which edge nodes will request your origin content. Azion appends the **Path** to the URI when forwarding the request to your origin, but it doesn't append the URI for user requests. For instance, if you determine that the path of your origin is under `/secure` and you have configured your application to be delivered through `yourdomain.com`, the `/secure` path won't be part of the request to the end user.

:::caution
The definition of an **Origin Path** is optional. If not specified, only the root URI will be considered.
:::

#### Address

To define an origin for your content, you need to specify the address of your server as either a domain name in FQDN format or an IPv4/IPv6 address. If you have defined the **Origin Protocol Policy** as **Enforce HTTP** or **Enforce HTTPS**, you can customize the origin port using the notation `host:port`.

#### Protocol Policy

Azion's delivery architecture offers you the flexibility to tailor the type of connection between the edge nodes and your origin based on your specific requirements:

> - **Preserve HTTP/HTTPS protocol**: this option maintains the same connection protocol (*HTTP or HTTPS*) and ports used by your users when accessing your content on Azion. It ensures seamless continuity from Azion to your origin server.

Reviewer: seamless?

- **Enforce HTTP**: with this setting, the connection between Azion's edge nodes and your origin will be exclusively through HTTP, regardless of the connection protocol and ports used by your users to access Azion's content. You can customize a specific port for your origin in the **Address** field, deviating from the default port (80 for HTTP) if desired.
- **Enforce HTTPS**: this option mandates the connection between Azion's edge nodes and your origin to be exclusively through HTTPS, regardless of the connection protocol and ports used by your users to access Azion's content. You can customize a specific port for your origin in the **Address** field, different from the default port (443 for HTTPS) if required.

#### HMAC Authentication 

> For **Single Origin**, if the origin is located in an object storage with private access, you must provide HMAC authentication credentials. When enabling HMAC authentication, your private content will begin to be delivered to your users through **Edge Application**. 

Reviewer 2: @LuizaVSantos : HMAC authentication deve especificar no texto apenas o funcionamento do recurso. Não precisa citar Edge application aqui.

HMAC (Hash-Based Message Authentication Codes) built-in support for AWS S3 or compatible authentication.
Enabling this will affect all addresses used for the Edge Connector. Individual credentials for specific addresses are not supported in this version.

You should include in the HMAC access credentials given by your storage provider: **Region**, **Access Key**, and **Secret Key**. 

:::note
**Secret Key** is protected by being displayed only to its users with write permission and maintaining their access security.
:::

#### Timeouts

> Displays timeout setting in seconds pre-defined by Azion. To customize timeout values, [contact the support team](https://tickets.azion.com/en/support/login).

Reviewer 2: @LuizaVSantos Adicione à seção aqui uma adpatação para quais são os timeouts e uma breve explicação deles (tirado do texto abaixo)

- `connection_timeout` | Timeout in seconds for establishing a connection to the origin.
- `read_write_timeout` | Timeout in seconds for read/write operations, that is the maximum interval allowed for the last seen byte sent or received on the origin active connection.

REMOVA a questão de customização de timeout via Support, pois isso será configurável na interface, pelo cliente.

#### Modules

##### Load Balancer

When **HTTP** is selected as an connector type, you can configure the **Balancing Method** to determine how large amounts of requests are distributed among addresses, add multiple **Addresses** to your Edge Connector, and change the **Sever Role** and **Weight** of each address.

<LinkButton link="/en/documentation/products/build/edge-application/load-balancer/" label="Learn more about Load Balancer" />

##### Origin Shield

> Origin Shield provides a list of IP/CIDR addresses used exclusively by Azion in its distributed infrastructure, ensuring that only trusted Azion IPs are allowed access. The list is maintained by Azion and made available via Azion Console and API. Clients are responsible for automating updates to their security policies to keep them aligned with the current Origin Shield addresses.

Reviewer 2: @LuizaVSantos Aqui podemos deixar a instrução mais clara.
Sugestão:

Origin Shield is a security feature that provides a dynamic Network List, "Azion Origin Shield," containing all IP/CIDR addresses used by Azion's network infrastructure. This list serves as a data source for enforcing a strict IP Access Control List (ACL) on your origin servers. By configuring your origin's firewall to permit inbound traffic exclusively from the addresses specified in this list, you establish a Layer 3/4 security perimeter, ensuring that only Azion's edge servers can connect to your origin and blocking all direct access from external sources.
lients are responsible for automating updates to their security policies to keep them aligned with the current Origin Shield addresses.
To activate, enable Origin IP ACL for at least one Edge Connector in your account.

| `config` | Load balancer configuration options. | – |
| `method` | Load balancing method. | `"round_robin"` |
| `max_retries` | Maximum number of retries for backend connections. | `0` |
| `connection_timeout` | Timeout in seconds for establishing a connection. | `60` |
| `read_write_timeout` | Timeout in seconds for read/write operations. | `120` |
| `origin_shield` | Origin Shield settings. | – |
| `enabled` | Indicates if Origin Shield is enabled. | `false` |
| `config` | Origin Shield configuration options. | – |
| `origin_ip_acl` | IP ACL configuration for Origin Shield. | – |
| `enabled` | Enables IP ACL for Origin Shield. | `false` |
| `hmac` | HMAC configuration for Origin Shield. | – |
| `enabled` | Indicates if HMAC security is enabled. | `false` |
| `config` | HMAC configuration options. | – |
| `type` | HMAC type. | `"aws4_hmac_sha256"` |
| `attributes` | HMAC credential set. | – |
| `region` | AWS region for HMAC. | `"string"` |
| `service` | AWS service for HMAC. | `"s3"` |
| `access_key` | AWS access key for HMAC. | `"string"` |
| `secret_key` | AWS secret key for HMAC. | `"string"` |

---

### Edge Storage

> When **Edge Storage** is selected as an origin type, you must add the **Bucket Name** and **Prefix**.

Reviewer 2: @LuizaVSantos 
Vale deixar claro que esse conector é do produto Edge Storage.

Sugestão:
When **Edge Storage** is selected as an origin type, it connects directly with an Azion Edge Storage [link para doc do produto] bucket where you must add the **Bucket Name** and **Prefix**.

## Implementation

| Scope | Resource |
| --- | --- |
| Edge Application first steps | [First steps](/en/documentation/products/build/edge-application/first-steps/#origins) |
| Load Balancer module | [Load Balancer](/en/documentation/products/build/edge-application/load-balancer/) |

---

## Connector Type

There are three options for you to connect your origin:

- **HTTP**: Connect to external origins via secure HTTP protocol for data delivery.
- **Edge Storage**: Link your Edge Storage bucket.
- **Live Ingest**: Designed for live video stream ingestion, this feature allows you to select the ingestion region to optimize latency.

### HTTP

Allows you to connect to generic HTTP servers, offering several configuration options, including:

- Connection Options: Provides control over DNS resolution, transport policy (HTTP/HTTPS), HTTP version, host, path prefix, redirect behavior, and headers for real IP and port.

> - Advanced Modules: Enables features such as load balancing when using  multiple (using methods like round-robin, least connections, IP hash, and support for retries and timeouts), Origin Shield (with IP ACL), and HMAC for additional security when communicating with the backend.

Reviewer 3: - Advanced Modules: Enables features such as load balancing when using multiple (using methods like round-robin, least connections, IP hash, and support for retries and timeouts), Origin Shield (with IP ACL), and HMAC for additional security when communicating with the backend.

Reviewer: Não seria "when using multiple methods like round-robin, least connections, IP hash, and support for retries and timeouts, Origin Shield..."?

- **Enforce HTTPS**: this option mandates the connection between Azion's edge nodes and your origin to be exclusively through HTTPS, regardless of the connection protocol and ports used by your users to access Azion's content. You can customize a specific port for your origin in the **Address** field, different from the default port (443 for HTTPS) if required.

#### HMAC Authentication 

HMAC (Hash-Based Message Authentication Codes) is the built-in support for AWS S3 or compatible authentication.

You should include in the HMAC access credentials given by your storage provider: **Region**, **Access Key**, and **Secret Key**. 

:::note
Enabling this will affect all addresses used for the Edge Connector. Individual credentials for specific addresses are not supported in this version.
:::

#### Timeouts

The `connection_timeout` parameter specifies the timeout, in seconds, for establishing a connection to the origin. The `read_write_timeout` parameter defines the timeout, in seconds, for read and write operations. This value represents the maximum interval allowed between the last byte sent or received on the active connection to the origin.

#### Modules

##### Load Balancer

> When **HTTP** is selected as an connector type, you can configure the **Balancing Method** to determine how large amounts of requests are distributed among addresses, add multiple **Addresses** to your Edge Connector, and change the **Sever Role** and **Weight** of each address.

Reviewer: ```suggestion
When **HTTP** is selected as a connector type, you can configure the **Balancing Method** to determine how large amounts of requests are distributed among addresses, add multiple **Addresses** to your Edge Connector, and change the **Server Role** and **Weight** of each address.
```

<LinkButton link="/en/documentation/products/build/edge-application/load-balancer/" label="Learn more about Load Balancer" />

##### Origin Shield

**Origin Shield** is a security feature that provides a dynamic Network List, `Azion Origin Shield`, containing all IP/CIDR addresses used by Azion's network infrastructure. 

This list serves as a data source for enforcing a strict IP Access Control List (ACL) on your origin servers. By configuring your origin's firewall to permit inbound traffic exclusively from the addresses specified in this list, you establish a Layer 3/4 security perimeter, ensuring that only Azion's edge servers can connect to your origin and blocking all direct access from external sources.

Clients are responsible for automating updates to their security policies to keep them aligned with the current **Origin Shield** addresses.

> To activate, enable Origin IP ACL for at least one Edge Connector in your account.

Reviewer: ```suggestion
To activate it, enable Origin IP ACL for at least one Edge Connector in your account.
```

> #### Atributes

Reviewer: ```suggestion
#### Attributes
```

#### Attributes

Below are the attributes for the [Edge Connector API](https://api.azion.com/v4#/operations/GetEdgeConnectorConnectors) when using the Edge Storage connector type.

| Attribute | Type   | Description                     |
|-----------|--------|---------------------------------|
| bucket    | String | The name of the bucket.         |
| prefix    | String | The prefix path for the bucket. |

<LinkButton link="/en/documentation/products/store/edge-storage/" label="Go to Edge Storage reference" />

### Live Ingest

This connector type can be used for scenarios where content providers require high fidelity and low latency, such as live events, e-sports, and educational content. It supports ingestion of live video streams and enables selection of the ingestion region to optimize latency. The platform uses a network of edge locations to improve delivery efficiency and scalability, and provides security features for stream protection. To use this connector type, select your preferred ingestion region to minimize latency, start streaming your live video using the appropriate endpoint, and monitor performance to ensure optimal delivery.

> <LinkButton link="(/en/documentation/architectures/live-streaming-delivery/live-streaming-delivery-with-hls/" label="Go to Live Ingest Solution" />

Reviewer 3: <LinkButton link="/en/documentation/architectures/live-streaming-delivery/live-streaming-delivery-with-hls/" label="Go to Live Ingest Solution" />

Reviewer 3: tem um parêntese antes do link: link="(/en/...

> <LinkButton link="(/en/documentation/architectures/live-streaming-delivery/live-streaming-delivery-with-hls/" label="Go to Live Ingest Solution" />

Reviewer: ```suggestion
<LinkButton link="/en/documentation/architectures/live-streaming-delivery/live-streaming-delivery-with-hls/" label="Go to Live Ingest Solution" />
```
